backend/lucene/src/main/java/org/hibernate/search/backend/lucene/resources/impl/BackendThreads.java

```diff
@@ -6,53 +6,65 @@
  */
 package org.hibernate.search.backend.lucene.resources.impl;
 
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.ScheduledExecutorService;
-
-import org.hibernate.search.backend.lucene.cfg.LuceneBackendSettings;
-import org.hibernate.search.engine.cfg.spi.ConfigurationProperty;
+import org.hibernate.search.engine.backend.work.execution.spi.BackendWorkExecutorProvider;
 import org.hibernate.search.engine.cfg.ConfigurationPropertySource;
-import org.hibernate.search.engine.cfg.spi.OptionalConfigurationProperty;
+import org.hibernate.search.engine.cfg.spi.BackendSpiSettings;
+import org.hibernate.search.engine.cfg.spi.ConfigurationProperty;
+import org.hibernate.search.engine.common.execution.SimpleScheduledExecutor;
+import org.hibernate.search.engine.environment.bean.BeanReference;
+import org.hibernate.search.engine.environment.bean.BeanResolver;
 import org.hibernate.search.engine.environment.thread.spi.ThreadPoolProvider;
 import org.hibernate.search.engine.environment.thread.spi.ThreadProvider;
 import org.hibernate.search.util.common.AssertionFailure;
 import org.hibernate.search.util.common.impl.Closer;
 
 public class BackendThreads {
 
-	private static final OptionalConfigurationProperty<Integer> THREAD_POOL_SIZE =
-			ConfigurationProperty.forKey( LuceneBackendSettings.THREAD_POOL_SIZE )
-					.asIntegerStrictlyPositive()
+	private static final ConfigurationProperty<BeanReference<? extends BackendWorkExecutorProvider>> BACKEND_WORK_EXECUTOR_PROVIDER =
+			ConfigurationProperty.forKey( BackendSpiSettings.Radicals.BACKEND_WORK_EXECUTOR_PROVIDER )
+					.asBeanReference( BackendWorkExecutorProvider.class )
+					.withDefault( LuceneBackendWorkExecutorProvider.BACKEND_WORK_EXECUTOR_PROVIDER )
 					.build();
-
 	private final String prefix;
 
 	private ThreadPoolProvider threadPoolProvider;
-	private ScheduledExecutorService writeExecutor;
+	private SimpleScheduledExecutor writeExecutor;
 
 	public BackendThreads(String prefix) {
 		this.prefix = prefix;
 	}
 
-	public void onStart(ConfigurationPropertySource propertySource, ThreadPoolProvider threadPoolProvider) {
+	public void onStart(ConfigurationPropertySource propertySource, BeanResolver beanResolver, ThreadPoolProvider threadPoolProvider) {
 		if ( this.writeExecutor != null ) {
 			// Already started
 			return;
 		}
 		this.threadPoolProvider = threadPoolProvider;
 
-		int threadPoolSize = THREAD_POOL_SIZE.get( propertySource )
-				.orElse( Runtime.getRuntime().availableProcessors() );
-		// We use a scheduled executor for write so that we perform all commits,
-		// scheduled or not, in the *same* thread pool.
-		this.writeExecutor = threadPoolProvider.newScheduledExecutor(
-				threadPoolSize, prefix + " - Worker thread"
-		);
+
+		BackendWorkExecutorProvider provider = BACKEND_WORK_EXECUTOR_PROVIDER.getAndTransform(
+				propertySource, beanResolver::resolve ).get();
```

Comment:
`BeanHolder` is closable and should be closed; please see how beanholders are handled elsewhere, for example there:

https://github.com/hibernate/hibernate-search/blob/2dbdfaad4b4b75483091f7d9485ed5e1d2c77e34/backend/elasticsearch-aws/src/main/java/org/hibernate/search/backend/elasticsearch/aws/impl/ElasticsearchAwsHttpClientConfigurer.java#L93-L96